### PR TITLE
Feature: add immutable version of `forNode` & `forNodePos`

### PR DESCRIPTION
**Pull request by ynfle**
This is for both `forNode` and `forNodePos`

### CODE REVIEW

**Comment by PMunch** [comment]
Hmm, do we really need this `forPos` to return the original tree?

**Comment by ynfle** [comment]
We definitely don't need it to return but it's nice for
```nim
node.forNode(nnkIdent, x => echo x)
    .forNode(nnkTypeDef, x => echo x[0])
```
I don't really have a strong opinion. What ever you think.

**Comment by ynfle** [comment]
@PMunch Just bumping. I'm fine with either option

**Comment by PMunch** [comment]
Right, I think I'll drop the chaining for the non-modifying version, seems a bit odd to have it there. Will merge and do some minor changes tomorrow.

**Comment by ynfle** [comment]
Updated

File: src/macroutils.nim [hunk 1]
```diff
@@ -1057,6 +1074,26 @@ template forNode*(node: NimNode,
...
   forNode(node, kind, action, 0, int.high)
 
+proc forNode*(node: NimNode,
+              kind: NimNodeKind or NimNodeKinds,
+              action: proc (x: NimNode),
+              depth = 0, maxDepth = int.high) =
+  ## Takes a NimNode and a NimNodeKind (or a set of NimNodeKind) and applies
+  ## the procedure passed in as `action` to every node that matches the kind
+  ## throughout the tree. The `x` passed to `action` is the node in the tree.
+  ## NOTE: This does not modify the original node tree and returns for easy
```
Comment (PMunch) [inline src/macroutils.nim:1084]: You forgot to remove this comment when you rewrote it to not return anything

**Pull request merged**